src/Stryker.Core/Stryker.Core/Compiling/FsharpCompilingProcess.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using FSharp.Compiler.SourceCodeServices;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.Extensions.Logging;
+using Microsoft.FSharp.Collections;
+using Microsoft.FSharp.Control;
+using Stryker.Core.Exceptions;
+using Stryker.Core.Initialisation.Buildalyzer;
+using Stryker.Core.Logging;
+using Stryker.Core.MutationTest;
+using ParsedInput = FSharp.Compiler.SyntaxTree.ParsedInput;
+
+namespace Stryker.Core.Compiling
+{
+    public class FsharpCompilingProcess
+    {
+        private readonly MutationTestInput _input;
+        private readonly ILogger _logger;
+        private readonly IFileSystem _fileSystem;
+
+        public FsharpCompilingProcess(MutationTestInput input,
+            IRollbackProcess rollbackProcess, IFileSystem fileSystem)
+        {
+            _input = input;
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<FsharpCompilingProcess>();
+            _fileSystem = fileSystem;
+        }
+
+        private string AssemblyName =>
+            _input.ProjectInfo.ProjectUnderTestAnalyzerResult.GetAssemblyName();
+
+        public CompilingProcessResult Compile(IEnumerable<ParsedInput> syntaxTrees, bool devMode)
+        {
+            var analyzerResult = _input.ProjectInfo.ProjectUnderTestAnalyzerResult;
+
+            FSharpList<ParsedInput> trees = ListModule.OfSeq(syntaxTrees.Reverse());
+            FSharpList<string> dependencies = ListModule.OfSeq(analyzerResult.References);
+
+            //we need a checker if we want to compile 
+            var checker = FSharpChecker.Create(projectCacheSize: null, keepAssemblyContents: null, keepAllBackgroundResolutions: null, legacyReferenceResolver: null, tryGetMetadataSnapshot: null, suggestNamesForErrors: null, keepAllBackgroundSymbolUses: null, enableBackgroundItemKeyStoreAndSemanticClassification: null);
+
+            var pathlist = new List<string>();
+            var pdblist = new List<string>();
```

Comment:
This variable tells me nothing about what it contains.

Reply:
pdb's aka portable debug binaries. It's a well known dotnet term.